### PR TITLE
fix(ZUR-7): fix and Improve search functionality 

### PR DESCRIPTION
**Pull request by jeckdix**
# Added the ability to view search results by type and improved the overall search component UI

**My PR Fixes ID: ZUR-7 for team clutch (Frontend)

# Changes proposed

## What were you told to do?
I was required to make the search functionality work

## What did you do?

- Added the ability to view search results by type (added tabs)
- Created different cards for search type
- Improved the result card UI
- Fixed styling issues for responsiveness
- Removed redundant lines of code

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos
https://www.loom.com/share/d5245f06dbd040538f1d4d04bc6cd7e7

### CODE REVIEW

**Comment by jeckdix** [comment]
@JulianaSau This is the new PR with clean commit history for search functionality. I have closed the previous one.

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx [hunk 1]
```diff
@@ -31,52 +38,73 @@ const TopBarSearchModal = () => {
...
+      //       setResult(response.data.results.data);
+      //     }
+      //     setLoading(false);
+      //   } catch (e) {
+      //     setLoading(false);
+      //     console.error(e);
+      //   }
+      // };
+      // getResult();
+
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:67]: Remove this commented out code if its not being used

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx [hunk 2]
```diff
@@ -31,52 +38,73 @@ const TopBarSearchModal = () => {
...
-      try {
-        const response = await exactPlugin.filterCall(user.org_id, user._id);
+  // useEffect(() => {
+  //   async function getData() {
+  //     if (!exactPlugin?.filterCall) {
+  //       setFilters({});
+  //       return;
+  //     }
+  //     try {
+  //       const response = await exactPlugin.filterCall(user.org_id, user._id);
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:97]: Same to this code

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx [hunk 3]
```diff
@@ -31,52 +38,73 @@ const TopBarSearchModal = () => {
...
-  }, [exactPlugin?.name, user._id]);
+  //       if (response.status >= 200 || response.status <= 299) {
+  //         setFilters(response.data.data);
+  //       }
+  //     } catch (e) {
+  //       setFilters({});
+  //     }
+  //   }
+  //   getData();
+  // }, [exactPlugin?.name, user._id]);
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:107]: Same to this

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx [hunk 4]
```diff
@@ -113,7 +141,7 @@ const TopBarSearchModal = () => {
         )}
         <input
           type="text"
-          placeholder={t("search.placeholder")}
+          placeholder={t("Search")}
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:144]: Why are you changing this. This is important for the translation team who added this there

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx [hunk 5]
```diff
@@ -145,7 +173,7 @@ const TopBarSearchModal = () => {
               <input
                 type="text"
                 className={styles._MainInput}
-                placeholder={t("search_placeholder")}
+                placeholder={t("Search")}
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:176]: same to this. Don't edit it out. It was added there for a reason

File: yarn.lock [hunk 6]
```diff
@@ -5361,10 +5361,7 @@ agora-rtc-react@^1.1.0:
   dependencies:
     agora-rtc-sdk-ng "^4.3.0"
 
-
 agora-rtc-sdk-ng@^4.15.0, agora-rtc-sdk-ng@^4.3.0:
-agora-rtc-sdk-ng@^4.3.0:
```
Comment (JulianaSau) [inline yarn.lock:5366]: Why is your PR editing the agora installation?

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: packages/main/src/components/protected/topbar/components/BigModal.jsx [hunk 7]
```diff
@@ -33,13 +36,19 @@ export const BigModal = ({
   }, [isLoadingUp]);
   let i = 0;
 
-  const sidebar = document.getElementById(
-    "single-spa-application:@zuri/sidebar"
-  );
+  const getResultCount = type => {
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/BigModal.jsx:39]: Why did you remove the code above?

File: packages/main/src/components/protected/topbar/components/BigModal.jsx [hunk 8]
```diff
@@ -6,23 +6,26 @@ import Styled from "styled-components";
...
-  const newName = pluginName.split("/");
+  // const pluginName = window.location.href;
+  // const newName = pluginName.split("/");
 
-  const exactPlugin = plugins.find(
-    plugin => newName[3] === plugin.name || newName[3] === plugin.name + "#"
-  );
+  // const exactPlugin = plugins.find(
+  //   plugin => newName[3] === plugin.name || newName[3] === plugin.name + "#"
+  // );
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/BigModal.jsx:29]: Why did you remove this code?

File: packages/main/src/components/protected/topbar/components/BigModal.jsx [hunk 9]
```diff
@@ -6,23 +6,26 @@ import Styled from "styled-components";
...
+  searchType
 }) => {
   const [results, setResult] = useState(result);
   const [isLoading, setLoading] = useState(isLoadingUp);
+  const [searchFocus, setSearchFocus] = useState(searchType);
 
-  const pluginName = window.location.href;
-  const newName = pluginName.split("/");
+  // const pluginName = window.location.href;
+  // const newName = pluginName.split("/");
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/BigModal.jsx:25]: same to this

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx (hunk 1 above)
Comment (jeckdix) [reply packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:67]: It's not been used for now but will be used when the search endpoint is ready. 
I will remove it.

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx (hunk 2 above)
Comment (jeckdix) [reply packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:97]: Reason same as above. I will remove it

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx (hunk 4 above)
Comment (jeckdix) [reply packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:144]: My bad please. I will revert the change.

File: packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx (hunk 5 above)
Comment (jeckdix) [reply packages/main/src/components/protected/topbar/components/TopBarSearchModal.jsx:176]: Noted

File: packages/main/src/components/protected/topbar/components/BigModal.jsx (hunk 9 above)
Comment (jeckdix) [reply packages/main/src/components/protected/topbar/components/BigModal.jsx:25]: The code is not in use right now and will break the app.

**Comment by jeckdix** [comment]
@JulianaSau  I have fixed the changes requested. Please review and merge.

**Review by JulianaSau — APPROVED** [review]
